packages/core/content-manager/server/src/history/controllers/history-version.ts

```diff
@@ -18,22 +41,26 @@ const createHistoryVersionController = ({ strapi }: { strapi: Strapi }) => {
         throw new errors.ForbiddenError('contentType and documentId are required');
       }
 
-      const params = ctx.query as HistoryVersions.GetHistoryVersions.Request['query'];
-
       /**
        * There are no permissions specifically for history versions,
        * but we need to check that the user can read the content type
        */
       const permissionChecker = getContentManagerService('permission-checker').create({
         userAbility: ctx.state.userAbility,
-        model: params.contentType,
+        model: ctx.query.contentType,
       });
 
       if (permissionChecker.cannot.read()) {
         return ctx.forbidden();
       }
 
-      const { results, pagination } = await getService(strapi, 'history').findVersionsPage(params);
+      const params: HistoryVersions.GetHistoryVersions.Request['query'] =
+        await permissionChecker.sanitizeQuery(ctx.query);
+
+      const { results, pagination } = await getService(strapi, 'history').findVersionsPage({
+        ...params,
+        ...validatePagination({ page: params.page, pageSize: params.pageSize }),
```

Comment:
I feel like the name of this function and the way it is used sort of conflict with each other. The name says it's validating but it's used here spreading a returned value.